src/DropdownMenu.tsx

```diff
@@ -109,22 +132,20 @@ const DropdownMenu = React.forwardRef(
 
     // For custom components provide additional, non-DOM, props;
     if (typeof Component !== 'string') {
-      menuProps.show = show;
-      menuProps.close = close;
-      menuProps.alignRight = alignEnd;
+      (menuProps as any).show = show;
+      (menuProps as any).close = close;
+      (menuProps as any).alignRight = alignEnd;
```

Comment:
I think we need to cast this to a more correct type vs casting to any below

Reply:
Yeah, I suppose, but how do we know the props of the custom `Component`? 🤔

Reply:
we don't, but that's not important here, the Component _must_ accept these props, e.g. the interface for whatever Component is, is "accepts at least these props". That said i think that's less important than just correctly typing menuProps so that it's safe to use _here_ not just so Component knows it.